ext/js/background/backend.js

```diff
@@ -480,8 +481,32 @@ class Backend {
             const noteIdsArray = await this._anki.findNoteIds(cannotAddNotes);
             for (let i = 0, ii = Math.min(cannotAdd.length, noteIdsArray.length); i < ii; ++i) {
                 const noteIds = noteIdsArray[i];
+
                 if (noteIds.length > 0) {
                     cannotAdd[i].info.noteIds = noteIds;
+
+                    /* TODO: should i be selecting the first ID like this?
+                        looking at findNoteIds I can't see how we could find
+                        more than one ID, but an example that proves me wrong
+                        would be cool (and also how to get the right ID) */
+                    const noteId = noteIds[0];
```

Comment:
Should probably return note info for all notes, i.e. just return an array that matches `noteIds`. Would probably be best to use the `'multi'` API of AnkiConnect (`findNoteIds` does this).

You probably don't need to worry about too much optimization, but the ideal situation is that only one API call is made. I can optimize that later if you like.